tools/pg_tools/macros.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0"?>
+<macros>
+    <xml name="inputs_outputs">
+        <inputs>
+            <param format="tabular" help="" label="Input tabular dataset" name="in_file" type="data" />
+            <yield />
+        </inputs>
+        <outputs>
+            <data format="tabular" name="out_file" label="${tool.name} on ${on_string}" />
+        </outputs>
+    </xml>
+
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@PG_VERSION@">postgresql</requirement>
+        </requirements>
+    </xml>
+
+    <token name="@PG_VERSION@">9.6.3</token>
+
+    <token name="@HELP_FOOTER@">
+    The postgresql tools, are using a postgresql database in user-space. Databases are packed into a tarball and stored in the user-history.
+    </token>
+
+    <token name="@UNTAR_INFILE@"><![CDATA[
+        tar -xJvf '$infile' &&
+        export PWD=`pwd` &&
+        sed -i.bak "s|unix_socket_directories.*|unix_socket_directories = '\$PWD/postgresql/'|" ./postgresql/db/postgresql.conf
+    ]]>
+    </token>
+
+    <token name="@PG_SETUP@"><![CDATA[
+            '$__tool_directory__/pglite' setup  -d ./postgresql &&
+            @PG_START@
+    ]]>
+    </token>
+
+    <token name="@PSQL@"><![CDATA[
+        psql "\$('$__tool_directory__/pglite' url -d ./postgresql)"
+    ]]>
+    </token>
+
+
+    <token name="@PG_START@"><![CDATA[
+        '$__tool_directory__/pglite' start  -d ./postgresql &&
+        sleep 10
```

Comment:
slightly worried about this. Wonder if we could switch a while loop?

Reply:
We could put the sleep in a while look and ask if the server is available? Not sure what the best way is here. Maybe we could make this start command blocking?

Reply:
I'd love a sleep in a while. Maybe something like the following would be good so we'd be safe from sleeping too much or too little, and we can be sure we won't sleep forever if something goes wrong.

```bash
exec_count=0
'$__tool_directory__/pglite' status | grep -F -q 'server is running'
while [ $? -ne 0 ]; do
    sleep 2
    exec_count=$((exec_count+1))
    if [[ "$exec_count" -gt 10 ]]; then exit 1; fi
    '$__tool_directory__/pglite' status | grep -F -q 'server is running'
done
```

(Completely untested.)

Reply:
I used, unless ... `status` still does not work reliably, but checking for the socket file in addition does the trick.